common/djangoapps/embargo/api.py

```diff
@@ -30,7 +30,10 @@ def redirect_if_blocked(course_key, access_point='enrollment', **kwargs):
     if settings.FEATURES.get('EMBARGO'):
         is_blocked = not check_course_access(course_key, **kwargs)
         if is_blocked:
-            return message_url_path(course_key, access_point)
+            if access_point != "enrollment":
+                if RestrictedCourse.is_disabled_access_check(course_key):
+                    return None
+                return message_url_path(course_key, access_point)
```

Comment:
In which scenario access will appear. we are getting either enrollment or courseware.